app/helpers/data_table_helper.rb

```diff
@@ -111,47 +116,132 @@ def render_data_table(data_table, options = {})
 
+  # To use this functionality, the following values must be set
+  # - call method in data_table -> store_source :filter_path (preferably in the base table)
+  # - set filter_class as fallback if you don't care about the query_class advantages
+  # - set custom_filter options on column or data_table
+  # - Example {
+  #         :custom_filter => {
+  #           :query_class  => FinancialQuery.to_s,
+  #           :query_filter => {
+  #             :class  => FinancialQueryFilter.to_s,
+  #             :params => {
+  #               :FinancialType => FinancialType::Composite.to_s
+  #             }
+  #           },
+  #           :type => 'enum'
+  #         }
+  #       }
+  # @param column Column
+  # @param data_table DataTable
+  def custom_filter(column, data_table)
+    filter_class = nil
+    query_class = nil
+    query_filter_class = nil
+    query_filter_params = nil
+    text_selector = nil
+    value_selector = nil
+    type = nil
+
+    if column.options[:custom_filter]
+      filter_class = column.options[:custom_filter][:filter_class]
+      query_class = column.options[:custom_filter][:query_class]
+      type = column.options[:custom_filter][:type]
+      text_selector = column.options[:custom_filter][:text_selector] || column.name
+      value_selector = column.options[:custom_filter][:value_selector] || column.name
+
+      if column.options[:custom_filter][:query_filter]
+        query_filter_class = column.options[:custom_filter][:query_filter][:class]
+        query_filter_params = column.options[:custom_filter][:query_filter][:params]
+      end
+    end
+
+    if data_table.options[:custom_filter]
+      filter_class ||= data_table.options[:custom_filter][:filter_class]
+      query_class ||= data_table.options[:custom_filter][:query_class]
+
+      if data_table.options[:custom_filter][:query_filter]
+        query_filter_class ||= data_table.options[:custom_filter][:query_filter][:class]
+        query_filter_params ||= data_table.options[:custom_filter][:query_filter][:params]
+      end
+    end
+
+    "
+        headerFilter: {
+            dataSource: {
+                paginate: false,
+                requireTotalCount: true,
+                map: function (dataItem) {
+                    return {
+                        text: dataItem['#{text_selector}'],
+                        value: dataItem['#{value_selector}']
+                    };
+                },
+                load: function (loadOptions) {
+                  var d = new $.Deferred();
+
+                  var params = {
+                    model: '#{filter_class}',
+                    field: '#{column.name}',
+                    query_class: '#{query_class}',
+                    query_filter_class: '#{query_filter_class}',
+                    params: #{(query_filter_params || {}).to_json},
+                    type: '#{type}',
+                  };
+
+                  var request = $.getJSON('#{data_table.store_url(self)}', params);
```

Comment:
column.options[:custom_filter][:store_url]

column.options[:custom_filter][:store_url].respond_to?(:call)
? column.options[:custom_filter][:store_url].call(self)
: self.send(column.options[:custom_filter][:store_url]

OR fallback to data_table.store_url(self)